File: py/barang.py
class Barang:
    def __init__(self, id=-1, nama="", harga="", merk="", kualitas=""):
        self.id = id
        self.nama = nama
        self.harga = harga
        self.merk = merk
        self.kualitas = kualitas

    # Setter & Getter
    def set_id(self, id): self.id = id
    def get_id(self): return self.id

    def set_nama(self, nama): self.nama = nama
    def get_nama(self): return self.nama

    def set_harga(self, harga): self.harga = harga
    def get_harga(self): return self.harga

    def set_merk(self, merk): self.merk = merk
    def get_merk(self): return self.merk

    def set_kualitas(self, kualitas): self.kualitas = kualitas
    def get_kualitas(self): return self.kualitas

    # Tampilkan data
    def tampilkan(self):
        print(f"ID       : {self.id}")
        print(f"Nama     : {self.nama}")
        print(f"Harga    : {self.harga}")
        print(f"Merk     : {self.merk}")
        print(f"Kualitas : {self.kualitas}")
        print("--------------------------")

File: php/barang.php
<?php
// =============================
// Class Barang
// merepresentasikan 1 buah barang
// dengan atribut: id, nama, harga, merk, kualitas, dan gambar
// =============================
class Barang {
    // properti private hanya bisa diakses lewat getter/setter
    private $id;
    private $nama;
    private $harga;
    private $merk;
    private $kualitas;
    private $gambar;

    // Konstruktor: dipanggil saat buat objek baru
    // contoh: new Barang("B01", "Laptop", 10000, "Asus", "A", "images/laptop.jpg");
    public function __construct($id, $nama, $harga, $merk, $kualitas, $gambar) {
        $this->id = $id;
        $this->nama = $nama;
        $this->harga = $harga;
        $this->merk = $merk;
        $this->kualitas = $kualitas;
        $this->gambar = $gambar;
    }

    // =============================
    // Getter (ambil nilai properti)
    // =============================
    public function getId() { return $this->id; }
    public function getNama() { return $this->nama; }
    public function getHarga() { return $this->harga; }
    public function getMerk() { return $this->merk; }
    public function getKualitas() { return $this->kualitas; }
    public function getGambar() { return $this->gambar; }

    // =============================
    // Setter (ubah nilai properti)
    // =============================
    public function setNama($nama) { $this->nama = $nama; }
    public function setHarga($harga) { $this->harga = $harga; }
    public function setMerk($merk) { $this->merk = $merk; }
    public function setKualitas($kualitas) { $this->kualitas = $kualitas; }
    public function setGambar($gambar) { $this->gambar = $gambar; }
}
?>
